src/docker.rs

```diff
@@ -23,29 +23,56 @@ pub enum Perm {
 
 pub struct RustEnv<'a> {
     pub args: &'a [&'a str],
-    pub privileged: bool,
     pub work_dir: (PathBuf, Perm),
     pub cargo_home: (PathBuf, Perm),
     pub rustup_home: (PathBuf, Perm),
     pub target_dir: (PathBuf, Perm),
 }
 
+pub struct MountConfig<'a> {
+    pub host_path: PathBuf,
+    pub container_path: &'a str,
+    pub perm: Perm,
+}
+
+impl<'a> MountConfig<'a> {
+    fn as_arg(&self) -> String {
```

Comment:
[`as_*` usually implies some sort of cost-free conversion](https://aturon.github.io/style/naming.html#conversions), maybe just `fn arg(&self) -> String`? i'm not sure, it's also not a big deal